main.go

```diff
@@ -0,0 +1,100 @@
+// A simple proxy that multiplexes a unix socket connection
+//
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/moby/moby/pkg/term"
+)
+
+func main() {
+	var (
+		debug         bool
+		logDir        string
+		root          string
+		agent         string
+		container     string
+		process       string
+		proxyExitCode bool
+		proxyStdio    bool
+		proxySignal   bool
+		proxyWinsize  bool
+	)
+
+	flag.BoolVar(&debug, "debug", false, "enable debug logging")
+	flag.StringVar(&logDir, "logdir", "", "the directory for the logging")
+	flag.StringVar(&root, "root", "", "root directory for storage of container state")
```

Comment:
What states do you have in mind? IMO the shim should be stateless. The way to restart is to start it again with the same parameters. But I can add states file if necessary.

Reply:
@bergwolf Yes, the shim should be stateless. But passing a `-root` option for `root directory for storage of container state` seems to indicate the opposite. Since this option is not used in the code, I'm not sure what the intention here is and I want to make sure the shim is indeed stateless.

Reply:
@sameo I agree. I'll drop it. It was needed by runv shim but not anymore here.